haystack/preprocessor/preprocessor.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import re
+from copy import deepcopy
+from functools import partial, reduce
+from itertools import chain
+from typing import List, Optional, Tuple, Generator, Set
+
+import nltk
+from more_itertools import windowed
+
+from haystack.preprocessor.base import BasePreProcessor
+
+logger = logging.getLogger(__name__)
+
+
+class PreProcessor(BasePreProcessor):
+    def __init__(
+        self,
+        clean_whitespace: Optional[bool] = True,
+        clean_header_footer: Optional[bool] = False,
+        clean_empty_lines: Optional[bool] = True,
+        split_by: Optional[str] = "passage",
+        split_size: Optional[int] = 10,
+        split_stride: Optional[int] = None,
+        split_mid_sentence: Optional[bool] = True,
+    ):
+        """
+        :param clean_header_footer: use heuristic to remove footers and headers across different pages by searching
+                                     for the longest common string. This heuristic uses exact matches and therefore
+                                     works well for footers like "Copyright 2019 by XXX", but won't detect "Page 3 of 4"
+                                     or similar.
+        :param clean_whitespace: strip whitespaces before or after each line in the text.
+        :param clean_empty_lines: remove more than two empty lines in the text.
+        :param split_by: split the document by "word", "sentence", or "passage". Set to None to disable splitting.
+        :param split_size: n number of splits to merge as a single document. For instance, if n -> 10 & split_by ->
+                           "sentence", then each output document will have 10 sentences.
```

Comment:
Is it possible to add the number of tokens here ?
Example case:
10 sentences per document, but one sentence is pretty large and has around 500 tokens ?
Then this one sentence is big enough that the rest of document is not in the focus of the reader modell.

Am I wrong ?

Reply:
If you decide to split by sentences (`split_by = "sentence"`), these cases can happen where document splits are larger than the max_seq_len of the reader. However, the reader will still "read" all of it as it can handle docs longer than max_seq_len with it's own sliding window. 
If you want to avoid this, you would probably split by 512 words via `split_by="word"`and `split_size=512` (or maybe tokens later) and set `split_mid_sentence=False`. This will basically split your texts into chunks of 512 words, but if the split is in the middle of a sentence it will not put that sentence in this chunk (and therefore have slightly less than 512 words).